pkg/hub/controllers/cluster/deregister.go

```diff
@@ -0,0 +1,302 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package cluster
+
+import (
+	"context"
+	"os"
+	"path"
+
+	hubv1alpha1 "github.com/kubeslice/apis/pkg/controller/v1alpha1"
+	"github.com/kubeslice/worker-operator/pkg/logger"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	serviceAccountName         = "cluster-deregister-sa"
+	cleanupContainer           = "cluster-cleanup"
+	clusterRoleName            = "cluster-deregister-role"
+	clusterRoleBindingName     = "cluster-deregister-rb"
+	clusterDeregisterConfigMap = "cluster-deregister-cm"
+	deregisterJobName          = "cluster-deregisteration-job"
+	operatorClusterRoleName    = "kubeslice-manager-role"
+)
+
+const (
+	scriptPath = "../../../../scripts"
+)
+
+// createDeregisterJob creates a job to uninstall worker-operator and notify controller
+// about registration status.
+func (r *Reconciler) createDeregisterJob(ctx context.Context, cluster *hubv1alpha1.Cluster) error {
+	log := logger.FromContext(ctx).WithName("cluster-deregister")
+	// Notify controller that the deregistration process of the cluster is in progress.
+	err := r.updateRegistrationStatusAndDeregisterInProgress(ctx, cluster, hubv1alpha1.RegistrationStatusDeregisterInProgress, nil)
+	if err != nil {
+		log.Error(err, "error updating status of deregistration on the controller")
+		return err
+	}
+	// Create a service account.
+	if err := r.MeshClient.Create(ctx, constructServiceAccount(), &client.CreateOptions{}); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			log.Info("service account already exists", "sa", serviceAccountName)
+		} else {
+			log.Error(err, "unable to create service account")
+			return err
+		}
+	}
+	// get operator clusterrole
+	operatorClusterRole, err := r.getOperatorClusterRole(ctx)
+	if err != nil {
+		log.Error(err, "unable to fetch operator clusterrole")
+		return err
+	}
+	// Create a cluster role.
+	if err := r.MeshClient.Create(ctx, constructClusterRole(operatorClusterRole), &client.CreateOptions{}); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			log.Info("cluster role already exists", "clusterrole", clusterRoleName)
+		} else {
+			log.Error(err, "unable to create cluster role")
+			return err
+		}
+	}
+	// Bind the service account with the cluster role.
+	if err := r.MeshClient.Create(ctx, constructClusterRoleBinding(), &client.CreateOptions{}); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			log.Info("cluster rolebinding already exists", "clusterrolebinding", clusterRoleBindingName)
+		} else {
+			log.Error(err, "unable to create cluster rolebinding")
+			return err
+		}
+	}
+	// get configmap data
+	data, err := getConfigmapData()
```

Comment:
getConfigmapData() is confusing. It could mean there is some configmap that needs to be read. To be clear, i would suggest changing the name to getCleanupScript(), and write a comment: 
// get the cleanup script in a long string and construct a configmap from it